File: api/app/song/domain/entity/Song.py
from datetime import datetime

from pydantic import BaseModel, Field
from ulid import ULID


class Song(BaseModel):
    id: ULID = Field(default_factory=lambda: ULID())
    title: str
    artist: str | None = None
    user_id: ULID
    created_at: datetime
    updated_at: datetime

File: console/src/domain/models/song.ts
import type { LyricsPart } from '../valueobjects/song'

export interface Song {
  id: string
  title: string
  artist: string | null
  userId: string
  createdAt: Date
  updatedAt: Date
}

/** One row from `GET /song/scrape-search-songs` (tuple: title, artist, lyrics). */
export interface ScrapeSearchSongHit {
  readonly title: string
  readonly artist: string | null
  readonly lyricsText: string | null
}

export interface ScrapeSearchSongsResult {
  readonly items: readonly ScrapeSearchSongHit[]
  readonly page: number
  readonly size: number
}

/** Preview-only result of `GET /song/lyrics/scrape` (not persisted). */
export interface ScrapeLyricsPreview {
  readonly matchedArtist: string
  readonly lyrics: readonly LyricsPart[]
}
